src/Analyzers/CSharp/Analyzers/UseCollectionExpression/CSharpUseCollectionExpressionForArrayDiagnosticAnalyzer.cs

```diff
@@ -88,14 +88,14 @@ private static void AnalyzeArrayInitializer(SyntaxNodeAnalysisContext context)
             : initializer;
 
         if (!UseCollectionExpressionHelpers.CanReplaceWithCollectionExpression(
-                semanticModel, arrayCreationExpression, cancellationToken))
+                semanticModel, arrayCreationExpression, skipVerificationForReplacedNode: false, cancellationToken))
```

Comment:
this is needed as the speculative system currently isn't ok with a node's kind changing this dramatically.  specifically from a method invocation to another expression form.  I wasn't comfortable mucking with the deep speculatrion logic, so easier to just not compare the bottommost nodes directly.

Reply:
😄I don't have any context about this, but choose to trust you here